raft/raft.go

```diff
@@ -611,6 +611,7 @@ func (r *raft) becomePreCandidate() {
 	// but doesn't change anything else. In particular it does not increase
 	// r.Term or change r.Vote.
 	r.step = stepCandidate
+	r.votes = make(map[uint64]bool)
```

Comment:
probably if pre-candidate times out, it needs to increase term to handle the potential vote spilt.